src/main/java/microsoft/exchange/webservices/data/AutodiscoverService.java

```diff
@@ -318,64 +323,64 @@ private void writeLegacyAutodiscoverRequest(String emailAddress,
-      URISyntaxException {
-    String url = String.format(AutodiscoverLegacyHttpUrl, "autodiscover."
-        + domainName);
+  private URI getRedirectUrl(String domainName)
+      throws EWSHttpException, XMLStreamException, IOException, ServiceLocalException, URISyntaxException {
+    String url = String.format(AutodiscoverLegacyHttpUrl, "autodiscover." + domainName);
 
-    this.traceMessage(TraceFlags.AutodiscoverConfiguration, String.format(
-        "Trying to get Autodiscover redirection URL from %s.", url));
+    traceMessage(TraceFlags.AutodiscoverConfiguration,
+        String.format("Trying to get Autodiscover redirection URL from %s.", url));
+
+    HttpWebRequest request = null;
 
-    HttpWebRequest request = new HttpClientWebRequest(this.getSimpleHttpConnectionManager());
     try {
-      request.setUrl(URI.create(url).toURL());
-    } catch (MalformedURLException e) {
-      String strErr = String.format("Incorrect format : %s", url);
-      throw new ServiceLocalException(strErr);
-    }
+      request = new HttpClientWebRequest(this.getSimpleHttpConnectionManager());
 
-    request.setAllowAutoRedirect(false);
-    request.setPreAuthenticate(false);
-    request.setRequestMethod("GET");
-    request.setUseDefaultCredentials(this.getUseDefaultCredentials());
-    if (!this.getUseDefaultCredentials()) {
-      ExchangeCredentials serviceCredentials = this.getCredentials();
-      if (null == serviceCredentials) {
-        throw new ServiceLocalException(Strings.CredentialsRequired);
+      try {
+        request.setUrl(URI.create(url).toURL());
+      } catch (MalformedURLException e) {
+        String strErr = String.format("Incorrect format : %s", url);
+        throw new ServiceLocalException(strErr);
       }
-      // Make sure that credentials have been authenticated if required
-      serviceCredentials.preAuthenticate();
 
-      // Apply credentials to the request
-      serviceCredentials.prepareWebRequest(request);
-    }
-    try {
-      request.prepareAsyncConnection();
-    } catch (Exception ex) {
-      ex.getMessage();
-      request = null;
-    }
+      request.setAllowAutoRedirect(false);
+      request.setPreAuthenticate(false);
+      request.setRequestMethod("GET");
+      request.setUseDefaultCredentials(this.getUseDefaultCredentials());
+      if (!this.getUseDefaultCredentials()) {
+        ExchangeCredentials serviceCredentials = this.getCredentials();
+        if (null == serviceCredentials) {
+          throw new ServiceLocalException(Strings.CredentialsRequired);
+        }
+        // Make sure that credentials have been authenticated if required
+        serviceCredentials.preAuthenticate();
+
+        // Apply credentials to the request
+        serviceCredentials.prepareWebRequest(request);
+      }
+
+      try {
+        request.prepareAsyncConnection();
+      } catch (Exception e) {
+        log.warn("Error while preparing asynchronous connection.", e);
```

Comment:
I would suggest 'error' in this case because the message underlines this. Maybe for documentation purpose we could add a section of 'how to declare logging and traces' to the readme file to also help others while logging with ews-java-api. Any comments?